packages/angular/ssr/src/manifest.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import { ApplicationRef, Type } from '@angular/core';
+import type { AngularServerApp } from './app';
+
+/**
+ * Manifest for the Angular server application engine, defining entry points.
+ */
+export interface AngularAppEngineManifest {
+  /**
+   * A record of entry points for the server application.
+   * Each entry consists of:
+   * - `key`: The base href.
+   * - `value`: A function that returns a promise resolving to an object containing the `AngularServerApp` type.
+   */
+  entryPoints: Record<string, () => Promise<{ AngularServerApp: typeof AngularServerApp }>>;
+
+  /**
+   * The base path for the server application.
+   */
+  basePath: string;
+}
+
+/**
+ * Manifest for a specific Angular server application, defining assets and bootstrap logic.
+ */
+export interface AngularAppManifest {
+  /**
+   * A record of assets required by the server application.
+   * Each entry consists of:
+   * - `key`: The path of the asset.
+   * - `value`: A function returning a promise that resolves to the file contents (default export).
+   */
+  assets: Record<string, () => Promise<{ default: string }>>;
```

Comment:
I am failing to understand how this would be correct, since the string `contents` would be in the default export, 

See: https://esbuild.github.io/try/#YgAwLjIzLjAAewogIGJ1bmRsZTogdHJ1ZSwKICBvdXRkaXI6ICcuLycsCiAgbG9hZGVyOiB7Jy5odG1sJzogJ3RleHQnfSwKfQBlAGVudHJ5LmpzAGV4cG9ydCBjb25zdCBhc3NldHMgPSB7J2Zvby5odG1sJzogKCk9PiBpbXBvcnQoJy4vaW5kZXguaHRtbCcpfQAAaW5kZXguaHRtbAA8aHRtbD48L2h0bWw+